draft-ietf-scitt-architecture.md

```diff
@@ -552,64 +581,131 @@ Editor's Note: This may be moved to appendix.
 POST <Base URL>/entries
 ~~~
 
+Headers:
+
+- `Content-Type: application/cose`
+
 Body: SCITT COSE_Sign1 message
 
 #### Response
 
 One of the following:
 
-- HTTP Status `201` - Registration was tentatively successful pending service consensus.
-- HTTP Status `400` - Registration was unsuccessful.
-  - Error code `AwaitingDIDResolutionTryLater`
-  - Error code `InvalidInput`
+- Status 201 - Registration is successful.
+  - Header `Location: <Base URL>/entries/<Entry ID>`
+  - Header `Content-Type: application/json`
+  - Body `{ "entryId": "<Entry ID"> }`
 
-[TODO] Use 5xx for AwaitingDIDResolutionTryLater
+- Status 202 - Registration is running.
+  - Header `Location: <Base URL>/operations/<Operation ID>`
+  - Header `Content-Type: application/json`
+  - (Optional) Header: `Retry-After: <seconds>`
+  - Body `{ "operationId": "<Operation ID>", "status": "running" }`
 
-The `201` response contains the `x-ms-ccf-transaction-id` HTTP header which can be used to retrieve the Registration Receipt with the given transaction ID. [TODO] this has to be made generic
+- Status 400 - Registration was unsuccessful due to invalid input.
+  - Error code `badSignatureAlgorithm`
+  - [TODO]: more error codes to be defined
 
-[TODO] probably a bad idea to define a new header, or is it ok? can we register a new one? https://www.iana.org/assignments/http-fields/http-fields.xhtml
+If 202 is returned, then clients should wait until registration succeeded or failed by polling the registration status using the Operation ID returned in the response. Clients should always obtain a receipt as a proof that registration has succeeded.
 
-The `400` response has a `Content-Type: application/json` header and a body containing details about the error:
+### Retrieve Operation Status
 
-```json
-{
-  "error": {
-    "code": "<error code>",
-    "message": "<message>"
-  }
-}
-```
+#### Request
 
-`AwaitingDIDResolutionTryLater` means the service does not have an up-to-date DID document of the DID referenced in the Signed Claims but is performing or will perform a DID resolution after which the client may retry the request. The response may contain the HTTP header `Retry-After` to inform the client about the expected wait time.
+~~~
+GET <Base URL>/operations/<Operation ID>
+~~~
 
-`InvalidInput` means either the Signed Claims message is syntactically malformed, violates the signing profile (e.g. signing algorithm), or has an invalid signature relative to the currently resolved DID document.
+#### Response
 
-### Retrieve Registration Receipt
+One of the following:
+
+- Status 200 - Registration is running
```

Comment:
Should we not return 202 here if the Registration is running ?

Reply:
No, status codes are always relative to the given resource. Here, 200 means the server was able to return the representation of the operation status. If it returned 202, it would mean that the server can't currently report on the status but will be able soon, which is not the intent here.